core/sdk/src/extn/client/extn_client.rs

```diff
@@ -240,10 +240,20 @@ impl ExtnClient {
                                 });
                             } else {
                                 // could be main contract
-                                Self::handle_stream(message, self.request_processors.clone());
+                                if !Self::handle_stream(
+                                    message.clone(),
```

Comment:
Is it possible to take ownership itself rather than cloning?

Reply:
It passes on the value down the processor in its own thread so it needs to get an instance. If we cant find a processor we send it to a different handler to inform the caller that there was no processor available to handle it.